packages/core/src/codewhisperer/service/securityScanHandler.ts

```diff
@@ -109,19 +111,32 @@ function mapRawToCodeScanIssue(issue: RawCodeScanIssue): CodeScanIssue {
     }
 }
 
-function mapToAggregatedList(codeScanIssueMap: Map<string, RawCodeScanIssue[]>, json: string) {
+export function mapToAggregatedList(
+    codeScanIssueMap: Map<string, RawCodeScanIssue[]>,
+    json: string,
+    editor: vscode.TextEditor | undefined,
+    scope: CodeWhispererConstants.CodeAnalysisScope
+) {
     const codeScanIssues: RawCodeScanIssue[] = JSON.parse(json)
-    codeScanIssues.forEach(issue => {
-        if (codeScanIssueMap.has(issue.filePath)) {
-            const list = codeScanIssueMap.get(issue.filePath)
-            if (list === undefined) {
-                codeScanIssueMap.set(issue.filePath, [issue])
-            } else {
-                list.push(issue)
-                codeScanIssueMap.set(issue.filePath, list)
+    const filteredIssues = codeScanIssues.filter(issue => {
+        if (scope === CodeWhispererConstants.CodeAnalysisScope.FILE && editor) {
+            for (let lineNumber = issue.startLine; lineNumber <= issue.endLine; lineNumber++) {
+                const line = editor.document.lineAt(lineNumber - 1)?.text
+                const codeContent = issue.codeSnippet.find(codeIssue => codeIssue.number === lineNumber)?.content
+                if (line !== codeContent) {
+                    return false
+                }
             }
+        }
+        return true
+    })
+
+    filteredIssues.forEach(issue => {
+        const filePath = issue.filePath
+        if (codeScanIssueMap.has(filePath)) {
+            codeScanIssueMap.get(filePath)!.push(issue)
```

Comment:
Can we avoid using `!`? I think it is not guaranteed that the map entry contains an array type